demos/door_manipulation_demo/src/door_open.cpp

```diff
@@ -0,0 +1,441 @@
+    }
+    return false;
+}   
+
+void sig_handler(int sig)
+{
+    g_caught_sigint = true;
+    ROS_INFO("caught sigint, init shutdown sequence...");
+    ros::shutdown();
+    exit(1);
+};
+
+
+moveit_msgs::GetPositionIK::Response computeIK(ros::NodeHandle n, geometry_msgs::PoseStamped p){
+    ros::ServiceClient ikine_client = n.serviceClient<moveit_msgs::GetPositionIK> ("/compute_ik");
+    
+    
+    moveit_msgs::GetPositionIK::Request ikine_request;
+    moveit_msgs::GetPositionIK::Response ikine_response;
+    ikine_request.ik_request.group_name = "arm";
+    ikine_request.ik_request.pose_stamped = p;
+    
+    /* Call the service */
+    if(ikine_client.call(ikine_request, ikine_response)){
+        ROS_INFO("IK service call success:");
+   
+    } else {
+        ROS_INFO("IK service call FAILED. Exiting");
+    }
+    
+    return ikine_response;
+}
+
+//get the recorded topics
+void joint_state_cb (const sensor_msgs::JointStateConstPtr& input) {
+    
+    if (input->position.size() == NUM_JOINTS){
+        current_state = *input;
+        heardJoinstState = true;
+    }
+}
+
+
+// Blocking call for user input
+void pressEnter(){
+    std::cout << "Press the ENTER key to continue";
+    while (std::cin.get() != '\n'){
+        std::cout << "Please press ENTER\n";
+    }    
+}
+
+//listen for where the arm is 
+void listenForArmData(float rate){
+    heardPose = false;
+    heardJoinstState = false;
+    ros::Rate r(rate);
+    while (ros::ok()){
+        ros::spinOnce();
+        
+        if (heardPose && heardJoinstState){
+            return;
+        }
+        r.sleep();
+    }
+}
+
+//get the first approach goal
+void goal_cb (const geometry_msgs::PoseStampedConstPtr& input)
+{
+    ROS_INFO("entered goal_cb");
+    first_goal.header = input->header;
+    first_goal.pose = input->pose; 
+    first_goal.pose.position.x -= .05;
+    second_goal.header = input->header;
+    second_goal.pose = input->pose;
+    second_goal.pose.position.x += .15;
+    heardGoal = true;
+    
+}
+void toolpos_cb (const geometry_msgs::PoseStampedConstPtr& input)
+{
+    ROS_INFO("entered toolposecb");
+    current_pose.header = input->header;
+    current_pose.pose = input->pose;
+}
+
+void plane_coeff_cb (const geometry_msgs::QuaternionConstPtr& input){
+    plane_coeff = *input;
+}   
+
+int main (int argc, char** argv)
+{   
+    // Initialize ROS
+    ros::init (argc, argv, "segbot_arm_door_open_detector");
+    ros::NodeHandle n;
+    tf::TransformListener listener;
+    //tested to be you of way of xtion camera for starting pose
+    start_pose.header.frame_id = "mico_link_base";
+    start_pose.pose.position.x = 0.14826361835;
+    start_pose.pose.position.y = -0.323001801968;
+    start_pose.pose.position.z = 0.233884751797;
+    start_pose.pose.orientation.x = 0.49040481699;
+    start_pose.pose.orientation.y = 0.468191160046;
+    start_pose.pose.orientation.z = 0.461722946003;
+    start_pose.pose.orientation.w = 0.571937124396;
```

Comment:
I see. Having a safe position for the arm out of the view of the xtion seems like it would be useful further down the line. Can you add it to the [joint space database](https://github.com/utexas-bwi/segbot_arm/blob/master/segbot_arm_manipulation/data/jointspace_position_db.txt)  and call the segbot_manipulation library to move it to that position?